clang/test/CheckedC/parsing/invalid-where-clause.c

```diff
@@ -5,19 +5,19 @@
 int f();
 
 void invalid_cases(_Nt_array_ptr<char> p, int a, int b) {
-  _Where ; // expected-error {{expected expression}}
-  _Where ;;;;; // expected-error {{expected expression}}
-  _Where _Where; // expected-error {{expected expression}} expected-error {{expected expression}}
-  _Where _And; // expected-error {{expected expression}} expected-error {{expected expression}}
-  _Where _And _And _And; // expected-error {{expected expression}} expected-error {{expected expression}} expected-error {{expected expression}} expected-error {{expected expression}}
+  _Where ; // expected-error {{expected bounds declaration or equality expression in where clause}}
+  _Where ;;;;; // expected-error {{expected bounds declaration or equality expression in where clause}}
+  _Where _Where; // expected-error {{expected bounds declaration or equality expression in where clause}} expected-error {{expected bounds declaration or equality expression in where clause}}
+  _Where _And; // expected-error {{expected bounds declaration or equality expression in where clause}} expected-error {{expected bounds declaration or equality expression in where clause}}
+  _Where _And _And _And; // expected-error {{expected bounds declaration or equality expression in where clause}} expected-error {{expected bounds declaration or equality expression in where clause}} expected-error {{expected bounds declaration or equality expression in where clause}} expected-error {{expected bounds declaration or equality expression in where clause}}
   _Where a; // expected-error {{expected comparison operator in equality expression}}
```

Comment:
Could this error message be improved too? Now that we have the WhereClause scope information, is it possible not to assume that `a` is the prefix of a potential equality expression?

Reply:
We enter `ParseBoundsExpression` only when we see an `identifier` followed by a `:`. Otherwise, we enter `ParseExpression`. According to this logic, we should assume that an `identifier` not followed by a `:` is the prefix of an equality expression. Hence, I think the error message is correct.

Reply:
This is now addressed in the latest change.